core/src/main/java/org/jdbi/v3/core/internal/SqlScriptParser.java

```diff
@@ -37,26 +47,37 @@ public String parse(CharStream charStream) {
         SqlScriptLexer lexer = new SqlScriptLexer(charStream);
         lexer.addErrorListener(new ErrorListener());
         boolean endOfFile = false;
+        int blockLevel = 0;
         while (!endOfFile) {
             Token t = lexer.nextToken();
             switch (t.getType()) {
                 case Token.EOF:
                     endOfFile = true;
                     break;
-                case SqlScriptLexer.SEMICOLON:
-                    semicolonHandler.handle(t, sb);
+                case SEMICOLON:
+                    if (blockLevel == 0) {
+                        semicolonHandler.handle(t, sb);
+                    } else {
+                        // preserve semicolons within begin/end block
+                        sb.append(t.getText());
+                    }
+                    break;
+                case BLOCK_BEGIN:
+                case BLOCK_END:
+                    blockLevel += BLOCK_BEGIN == t.getType() ? +1 : -1;
+                    sb.append(t.getText());
                     break;
-                case SqlScriptLexer.COMMENT:
-                case SqlScriptLexer.MULTI_LINE_COMMENT:
+                case COMMENT:
+                case MULTI_LINE_COMMENT:
                     break;
-                case SqlScriptLexer.NEWLINES:
+                case NEWLINES:
                     if (sb.length() > 0) {
-                        sb.append(' ');
+                        sb.append(System.lineSeparator()); // normalize newline sequences
```

Comment:
Hm. This is a bit odd - if you run a Jdbi client program on Windows, suddenly your Postgres server on Linux will see Windows line endings. Different text will be sent to the server depending on the client OS which I find a bit surprising.

Is it harmful to keep the existing `' '`, or we could use a fixed `'\n'`?

Reply:
Database servers are newline-agnostic. They must be, dealing with clients of all sorts and os.
I didn't like the super-long lines the original code creates. But sure, I can roll this change back.